### ISSUE TITLE
Change default merge strategy

### ISSUE THREAD

**Issue by relu91** [eclipse/thingweb.node-wot#558]
On PR #519 @erossignon [suggested](https://github.com/eclipse/thingweb.node-wot/pull/519#issuecomment-938353707) changing the default merge strategy from "Allow merge commits" to "Allow rebase merging". Rebasing would allow having a more linear history in the master branch where the current approach is to have merge commits of different PRs. I work with both approaches in different repositories and I have to admit I like both of them.

My take is that the current approach becomes not useful when you have _a lot of_ feature branches (a.k.a. PRs) in a short amount of time. I personally like the fact that you can understand from which PR a change was generated and use this information to look back in Github the discussion history. But of course, when you merge 20/30 Prs a week the thing is not really manageable. But this is not the case yet, looking at node-wot git graph I can still tell what's going on:
![image](https://user-images.githubusercontent.com/3170227/136756637-1220d93f-93ef-472e-8e46-0acb1bb6cb2d.png)

Please share you thoughts about this topic below :)

**Comment by egekorkan** [comment]
There is also this part in the Contributing.md about this : https://github.com/eclipse/thingweb.node-wot/blob/master/CONTRIBUTING.md#pull-requests-and-feature-branches

It is not exactly the same but some might be doing this while others are not. It would be good to say that rebase is required so that we have a single style of PRs? Not a strong opinion though :)

**Comment by egekorkan** [comment]
Also see https://github.com/eclipse/thingweb.node-wot/issues/559

**Comment by mkovatsc** [comment]
The guidelines combined what is used in other Eclipse projects and what we used for inner source.

A squash commit is good, when the PR is limited to a single feature or fix. A review before merging could ensure this. Then it could be as clean and modular as the rebase.

**Comment by danielpeintner** [comment]
@mkovatsc do you mean we should decide on a PR based manner?

**Comment by mkovatsc** [comment]
No, I meant being more strict with PRs (that they only add a single feature / fix a single bug) when changing to squash and merge.

**Comment by danielpeintner** [comment]
I totally agree with your statement. We tried to do that in the past.. and surely failed *sometimes* which gives room to improve.

Anyhow, even if a PR tackled just one aspect but was developed over several commits the history on master does not convey that. Hence, I prefer having a *single* commit that essentially says what was fixed/improved in a given PR and leads to s single entry in the history.

Note: projects like Java, JavaFX etc do that as well.

**Comment by relu91** [comment]
> Anyhow, even if a PR tackled just one aspect but was developed over several commits the history on master does not convey that. Hence, I prefer having a single commit that essentially says what was fixed/improved in a given PR and leads to s single entry in the history.

I'm ok with squashing too, as long as the other commits in the PR are:
- `chore`s of package JSON related with the changes of the PR
- Style fixes
- Reviews suggestions

I would not squash if:
- PR actually introduces more than on bugfix or feature
- Reviews ask to implement additional features
